plugins/experimental/buffer_upload/buffer_upload.cc

```diff
@@ -753,6 +753,15 @@ attach_pvc_plugin(TSCont /* contp ATS_UNUSED */, TSEvent event, void *edata)
         memcpy(replacement_host_str, host_hdr_str_val, host_hdr_str_val_len);
```

Comment:
Might as well clean up the memset.  You can set replacement_host_str[host_hdr_str_val_len + 1] = '\0' after the memcpy instead of doing a memset.